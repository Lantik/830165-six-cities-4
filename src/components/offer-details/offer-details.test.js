import React from 'react';
import renderer from 'react-test-renderer';
import OrderDetails from './offer-details.jsx';

const mock = {
  offer: {
    id: 1,
    price: 30,
    rating: 3.3,
    name: `Beautiful & luxurious apartment at great location`,
    type: `apartment`,
    inBookmarks: true,
    description: `Reprehenderit anim elit cillum officia ut proident dolore ea. Duis laborum elit aliqua reprehenderit ipsum exercitation consequat dolore incididunt. Do adipisicing deserunt do enim id proident velit enim minim in sit labore cupidatat. Nulla cupidatat excepteur id Lorem magna. Voluptate mollit dolor in nostrud qui elit laboris labore.`,
    photos: [`img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`],
    isPremium: true,
    bedroomsCount: 1,
    maxGuestsNumber: 3,
    conveniences: [`Wifi`, `Heating`, `Kitchen`, `Cable TV`],
    owner: {
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`,
      isSuper: true
    }
  }
};

it(`<OrderDetails/> component rendered correctly`, () => {
  const {offer} = mock;
  const tree = renderer
    .create(<OrderDetails
      offer={offer}
    />).toJSON();

  expect(tree).toMatchSnapshot();
});

