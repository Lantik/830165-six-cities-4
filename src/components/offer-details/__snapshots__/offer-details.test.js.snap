// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OfferDetails/> component rendered correctly 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="main.html"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-01.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-01.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-01.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-01.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-01.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-01.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Beautiful & luxurious apartment at great location
            </h1>
            <button
              className="property__bookmark-button button"
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "66%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              3.3
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              Apartment
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              1
               Bedroom
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              3
               adult
              s
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              30
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Wifi
              </li>
              <li
                className="property__inside-item"
              >
                Heating
              </li>
              <li
                className="property__inside-item"
              >
                Kitchen
              </li>
              <li
                className="property__inside-item"
              >
                Cable TV
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper--pro property__avatar-wrapper user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  src="img/avatar-angelina.jpg"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                Angelina
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                Reprehenderit anim elit cillum officia ut proident dolore ea. Duis laborum elit aliqua reprehenderit ipsum exercitation consequat dolore incididunt. Do adipisicing deserunt do enim id proident velit enim minim in sit labore cupidatat. Nulla cupidatat excepteur id Lorem magna. Voluptate mollit dolor in nostrud qui elit laboris labore.
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews · 
              <span
                className="reviews__amount"
              >
                2
              </span>
            </h2>
            <ReviewList
              reviews={
                Array [
                  Object {
                    "avatar": "img/avatar-angelina.jpg",
                    "date": 1237420800000,
                    "id": 2,
                    "rating": 20,
                    "text": "A quiet cozy and piscturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
                    "userName": "Katya",
                  },
                  Object {
                    "avatar": "img/avatar-max.jpg",
                    "date": 1242691200000,
                    "id": 3,
                    "rating": 99,
                    "text": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
                    "userName": "Helen",
                  },
                ]
              }
            />
            <form
              action="#"
              className="reviews__form form"
              method="post"
            >
              <label
                className="reviews__label form__label"
                htmlFor="review"
              >
                Your review
              </label>
              <div
                className="reviews__rating-form form__rating"
              >
                <input
                  className="form__rating-input visually-hidden"
                  id="5-stars"
                  name="rating"
                  type="radio"
                  value="5"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="5-stars"
                  title="perfect"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="4-stars"
                  name="rating"
                  type="radio"
                  value="4"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="4-stars"
                  title="good"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="3-stars"
                  name="rating"
                  type="radio"
                  value="3"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="3-stars"
                  title="not bad"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="2-stars"
                  name="rating"
                  type="radio"
                  value="2"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="2-stars"
                  title="badly"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="1-star"
                  name="rating"
                  type="radio"
                  value="1"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="1-star"
                  title="terribly"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
              </div>
              <textarea
                className="reviews__textarea form__textarea"
                id="review"
                name="review"
                placeholder="Tell how was your stay, what you like and what can be improved"
              />
              <div
                className="reviews__button-wrapper"
              >
                <p
                  className="reviews__help"
                >
                  To submit review please make sure to set 
                  <span
                    className="reviews__star"
                  >
                    rating
                  </span>
                   and describe your stay with at least 
                  <b
                    className="reviews__text-amount"
                  >
                    50 characters
                  </b>
                  .
                </p>
                <button
                  className="reviews__submit form__submit button"
                  disabled=""
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </section>
        </div>
      </div>
      <Map
        activeCoords={
          Array [
            52.3909553943508,
            4.85309666406198,
          ]
        }
        className="property__map"
        coords={
          Array [
            Array [
              52.369553943508,
              4.85309666406198,
            ],
            Array [
              52.3909553943508,
              4.929309666406198,
            ],
            Array [
              52.3809553943508,
              4.939309666406198,
            ],
          ]
        }
      />
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <NearbyOfferList
          offers={
            Array [
              Object {
                "bedroomsCount": 2,
                "conveniences": Array [
                  "Wifi",
                  "Heating",
                  "Kitchen",
                  "Cable TV",
                ],
                "coordinates": Array [
                  52.369553943508,
                  4.85309666406198,
                ],
                "description": "Reprehenderit anim elit cillum officia ut proident dolore ea. Duis laborum elit aliqua reprehenderit ipsum exercitation consequat dolore incididunt. Do adipisicing deserunt do enim id proident velit enim minim in sit labore cupidatat. Nulla cupidatat excepteur id Lorem magna. Voluptate mollit dolor in nostrud qui elit laboris labore.",
                "id": 2,
                "inBookmarks": true,
                "isPremium": true,
                "maxGuestsNumber": 2,
                "name": "Wood and stone place",
                "nearByOffers": Array [
                  1,
                  3,
                  4,
                ],
                "owner": Object {
                  "avatar": "img/avatar-angelina.jpg",
                  "isSuper": true,
                  "name": "Angelina",
                },
                "photos": Array [
                  "img/apartment-02.jpg",
                  "img/apartment-02.jpg",
                ],
                "price": 54,
                "rating": 3.8,
                "reviews": Array [
                  Object {
                    "avatar": "img/avatar-angelina.jpg",
                    "date": 1108771200000,
                    "id": 1,
                    "rating": 10,
                    "text": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
                    "userName": "Vlad",
                  },
                ],
                "type": "hotel",
              },
              Object {
                "bedroomsCount": 3,
                "conveniences": Array [
                  "Wifi",
                  "Kitchen",
                  "Cable TV",
                ],
                "coordinates": Array [
                  52.3909553943508,
                  4.929309666406198,
                ],
                "description": "Reprehenderit anim elit cillum officia ut proident dolore ea. Duis laborum elit aliqua reprehenderit ipsum exercitation consequat dolore incididunt. Do adipisicing deserunt do enim id proident velit enim minim in sit labore cupidatat. Nulla cupidatat excepteur id Lorem magna. Voluptate mollit dolor in nostrud qui elit laboris labore.",
                "id": 3,
                "inBookmarks": true,
                "isPremium": false,
                "maxGuestsNumber": 1,
                "name": "Stone and wood place",
                "nearByOffers": Array [
                  1,
                  4,
                ],
                "owner": Object {
                  "avatar": "img/avatar-angelina.jpg",
                  "isSuper": false,
                  "name": "Mike",
                },
                "photos": Array [
                  "img/apartment-03.jpg",
                ],
                "price": 400,
                "rating": 5,
                "reviews": Array [
                  Object {
                    "avatar": "img/avatar-max.jpg",
                    "date": 1324598400000,
                    "id": 0,
                    "rating": 70,
                    "text": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
                    "userName": "Max",
                  },
                  Object {
                    "avatar": "img/avatar.svg",
                    "date": 1108771200000,
                    "id": 4,
                    "rating": 10,
                    "text": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
                    "userName": "Vlad",
                  },
                ],
                "type": "room",
              },
              Object {
                "bedroomsCount": 4,
                "conveniences": Array [
                  "Kitchen",
                  "Cable TV",
                ],
                "coordinates": Array [
                  52.3809553943508,
                  4.939309666406198,
                ],
                "description": "Reprehenderit anim elit cillum officia ut proident dolore ea. Duis laborum elit aliqua reprehenderit ipsum exercitation consequat dolore incididunt. Do adipisicing deserunt do enim id proident velit enim minim in sit labore cupidatat. Nulla cupidatat excepteur id Lorem magna. Voluptate mollit dolor in nostrud qui elit laboris labore.",
                "id": 4,
                "inBookmarks": false,
                "isPremium": true,
                "maxGuestsNumber": 5,
                "name": "Luxurious & Beautiful apartment at great location",
                "nearByOffers": Array [
                  1,
                  2,
                ],
                "owner": Object {
                  "avatar": "img/avatar-angelina.jpg",
                  "isSuper": false,
                  "name": "John",
                },
                "photos": Array [
                  "img/apartment-small-03.jpg",
                  "img/apartment-small-03.jpg",
                  "img/apartment-small-03.jpg",
                ],
                "price": 10,
                "rating": 4.1,
                "reviews": Array [
                  Object {
                    "avatar": "img/avatar.svg",
                    "date": 1242691200000,
                    "id": 5,
                    "rating": 99,
                    "text": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
                    "userName": "Helen",
                  },
                  Object {
                    "avatar": "img/avatar-max.jpg",
                    "date": 1324598400000,
                    "id": 6,
                    "rating": 70,
                    "text": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
                    "userName": "Max",
                  },
                  Object {
                    "avatar": "img/avatar-angelina.jpg",
                    "date": 1108771200000,
                    "id": 7,
                    "rating": 10,
                    "text": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
                    "userName": "Vlad",
                  },
                ],
                "type": "house",
              },
            ]
          }
          onOfferHeaderClick={[Function]}
        />
      </section>
    </div>
  </main>
</div>
`;
